src/devices/cpu/dsp56156/inst.h

```diff
@@ -568,17 +544,13 @@ class Bcc_2: public Instruction
 		m_immediate = get_6_bit_signed_value(BITSn(word0,0x003f));
 		return true;
 	}
-	void disassemble(std::string& retString) const override
+	std::string disassemble() const override
 	{
-		std::string opcode = "b" + opMnemonicAsString(m_mnem);
-		// NEW // sprintf(opcode_str, "b.%s", M);
-
-		char temp[32];
-		if (m_immediate >= 0) sprintf(temp, "<*+$%x", m_immediate + 1);
-		else                  sprintf(temp, "<*-$%x", 1 - m_immediate - 2);
-		// NEW // sprintf(temp, "$%04x (%d)", pc + 1 + relativeInt, relativeInt);
+		auto opcode = "b" + opMnemonicAsString(m_mnem);
 
-		retString = opcode + " " + std::string(temp);
+		return opcode + (m_immediate >= 0 ?
+			util::string_format(" <*+$%x", m_immediate + 1) :
+			util::string_format(" <*-$%x", 1 - m_immediate - 2));
```

Comment:
Since you’re changing the whole function body, why not use a single `util::string_format` for each case (positive or negative displacement) and avoid the temporaries and concatenations?